lib/service/storage/elasticsearch.js

```diff
@@ -1192,6 +1192,18 @@ class ElasticSearch extends Service {
   async createCollection (index, collection, { mappings={}, settings={} } = {}) {
     this._assertValidIndexAndCollection(index, collection);
 
+    if (collection === HIDDEN_COLLECTION) {
+      throw kerror.get('collection_reserved', HIDDEN_COLLECTION);
+    }
+
+    if (await this.hasCollection(index, HIDDEN_COLLECTION, true)) {
+      try {
+        await this.deleteCollection(index, HIDDEN_COLLECTION);
+      }
+      catch (e) {
+        return this._esWrapper.reject(e);
```

Comment:
`_esWrapper.reject` is deprecated because it creates an unnecessary promise: being in an async function, throwing already returns a rejected promise.

```suggestion
        throw this._esWrapper.formatESError(e);
```